### PR TITLE
Make erroneous mail accounts not  selectable in composer modal

### LINKED ISSUE #8373: Erroneous mail account is selectable in composer modal
### Steps to reproduce

1. Add two accounts
2. Go to the database and copy the inbound_password from one account to another (the second one will now have an invalid password)
3. Reload the page
4. Click *New message*
5. Select *From*

### Expected behavior

The erroneous account is not selectable, but may be listed

### Actual behavior

The erroneous account is selectable. I can neither store a draft nor send a message. Any data I put into the composer is lost.

### Mail app version

3.2

### Mailserver or service

_No response_

### Operating system

_No response_

### PHP engine version

None

### Web server

None

### Database

None

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by hamza221**
fixes #8373

### CODE REVIEW

**Comment by hamza221** [comment]
cc @GretaD for the NcSelect

**Review by GretaD — APPROVED** [review]

**Review by miaulalala — APPROVED** [review]
lgtm didn't test

**Comment by hamza221** [comment]
@ChristophWurst Should I merge this?

File: lib/Service/AccountService.php [hunk 1]
```diff
@@ -181,4 +181,21 @@ public function updateSignature(int $id, string $uid, string $signature = null):
...
+	/**
+	 * @param string $currentUserId
+	 * @param int $accountId
+	 * @return bool
+	 */
+
+	public function testAccountConnection(string $currentUserId, int $accountId) :bool {
+		$account = $this->find($currentUserId, $accountId);
+		try {
+			$account->getImapConnection();
```
Comment (ChristophWurst) [inline lib/Service/AccountService.php:195]: ``\OCA\Mail\Account::getImapConnection`` is deprecated and should not be used anymore.

@hamza221 please refactor the code to use the IMAP client factory and add a unit test for this method.

Comment (hamza221) [reply lib/Service/AccountService.php:195]: #8962

**Pull request merged**